### PR TITLE
remove unused plugin for Spring

### PR DESCRIPTION
**Pull request by Netyyyy**
# Description

Remove unused plugin for Spring

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by saragluna — APPROVED** [review]

**Comment by Netyyyy** [comment]
Hi @mssfang @JonathanGiles @alzimmermsft @srnagar , could one of you help approve the changes in code-quality-reports?

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 1]
```diff
@@ -218,6 +218,11 @@
           "new": "method void com.azure.spring.cloud.resourcemanager.provisioning.ServiceBusProvisioner::provisionTopic(java.lang.String, java.lang.String, com.azure.spring.cloud.resourcemanager.provisioning.properties.ServiceBusTopicProperties)",
           "justification": "New method added to support more properties."
         },
+        {
+          "code": "java.method.addedToInterface",
+          "new": "method void com.azure.spring.cloud.appconfiguration.config.AppConfigurationRefresh::refreshAsync()",
+          "justification": "Method was added to an interface."
+        },
```
Comment (srnagar) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:225]: I do not see AppConfigurationRefresh changes in this PR. Also, this is a valid issue and should be fixed. We should not add new methods to existing interfaces. If required, we should add `default` methods.

**Review by srnagar — CHANGES_REQUESTED** [review]

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json (hunk 1 above)
Comment (mrm9084) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json:225]: @srnagar @saragluna I've added default methods in a but fix PR I have up https://github.com/Azure/azure-sdk-for-java/pull/41680.

Comment (srnagar) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json:225]: @mrm9084 I have added a few comments on the PR https://github.com/Azure/azure-sdk-for-java/pull/41680.

Comment (Netyyyy) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json]: @srnagar , I added Matt's changes here to make this pr merge first.
FYI @mrm9084

**Review by srnagar — APPROVED** [review]

**Pull request merged**